### PR TITLE
New overview page

### PR DESCRIPTION
**Pull request by r-czajkowski**
~Depends on: #2107 #2101~
Closes: #2126 #2002 

This PR updates the overview page and adds the empty sate pages. Here we also migrated data fetching and events subscription to redux, to improve data management between components.

obraz
obraz
obraz
obraz

### CODE REVIEW

**Comment by r-czajkowski** [comment]
fixed-net-status-and-wallet-menu

@liz-shinn fixed net status and wallet menu, thoughts? However, I think these elements should stay in a header.

**Comment by liz-shinn** [comment]
I will say that the network status would need to have the background element behind it that's in Figma if it's going to float, since there are sections where it won't have enough contrast with the background. It's the slightly transparent, white background.

Personally, I think it's nice to have just the wallet and status float like it's doing here. It keeps the useful elements of the header easily accessible, but doesn't take up a bunch of vertical screen real estate.

**Comment by liz-shinn** [comment]
Sidenote @r-czajkowski, the button for connect wallet could use some more padding. The border of the button is really tight around the icon and the text.

**Comment by r-czajkowski** [comment]
> I will say that the network status would need to have the background element behind

ah right- wrong gif. But all tiles are white, so the white background behind the status with `opacity: 0.6` prop, still doesn't look good.

overview-gif

**Comment by liz-shinn** [comment]
Ah, ok — perhaps an even more subtle opacity then, say 0.9, or could just do solid white.

I think that if we have floating elements, that it should be these two items. They're the two actionable items, so they'd be the things that you would want to be able to quicky click on without scrolling back to the top. But if we think that these floating elements look off, I don't think that means that the entire header should then float, since that's a huge part of the vertical screen real estate. Then it's just better to leave the header at the top of the screen. Curious @Shadowfiend and @Gluzman thoughts on this.

Overview page is looking great, though 🎉! Hooray!

File: solidity/dashboard/src/components/TokenOverview.jsx [hunk 1]
```diff
@@ -4,90 +4,169 @@ import * as Icons from "./Icons"
...
+      <ProgressBar
+        items={[{ value: staked, color: colors.primary }]}
+        total={totalBalance}
+        styles={{ margin: 0 }}
+      />
+      <span className="text-small text-grey-40">
+        {isConnected ? inPercentage : "-"}% Staked
+      </span>
+      <Link
+        to={{ pathname: "/delegate", hash: type }}
```
Comment (Shadowfiend) [inline solidity/dashboard/src/components/TokenOverview.jsx:164]: This link, and the other ones to staking from the overview, seems to be broken. Let's fix it before merging things to master.

**Review by Shadowfiend — APPROVED** [review]
Reviewing here, @liz-shinn it looks like the balance *number* should be darker (color-brand-mint-100) vs the keep logo (color-brand-mint-80), is that right?

Other things I noticed:
- There's a hand cursor on the KEEP amount, and it's abbreviated (e.g. 562k), vs the designs which use the fully-expanded number (e.g. 562,000). There's a localization component here, which we can probably handle by calling `toLocaleString()` on the number to get the right separators for thousands and decimals.

Going to have to accept the fact that I can't review 3k lines of changes haha. I left a few comments on a few things I noticed. Overall looks good, main flag is the Delegate buttons that are leading to a nonexistent URL (I believe it's delegate -> tokens/delegate).

**Comment by Shadowfiend** [comment]
> I think that if we have floating elements, that it should be these two items. They're the two actionable items, so they'd be the things that you would want to be able to quicky click on without scrolling back to the top. But if we think that these floating elements look off, I don't think that means that the entire header should then fix at the top and cover up screen content as you scroll, since that's a huge part of the vertical screen real estate. Then it's just better to leave the header at the top of the page. Curious @Shadowfiend and @Gluzman thoughts on this.

To reply here @liz-shinn , my take is that floating elements are rarely good. They mostly restrict what we can consider to be the viewport, so unless they provide constant value I don't think it's worth having them. For the connection widget we have, I personally don't think it brings constant value---but it's definitely in the category of “things we'd probably want to do user interviews to figure out for sure” in my mind. Someone who uses a lot of different accounts in Metamask, for example, might want to always be able to see what address they're viewing in the dApp itself. Not sure how many users are in that category though---I would guess not that many.

Anyway, that's a lot of guesswork, but short version is I think let's leave this not floating and maybe revisit it if we get some data that people are missing having that available at all times.

**Comment by liz-shinn** [comment]
Sounds good to me! Agreed that restricting the viewable area is not a great UX. No floating elements for the header 👍

**Comment by liz-shinn** [comment]
And yes, the balance _number_ should be the darker color. 

There's an accessibility issue with using the brand mint color for text — it doesn't have enough contrast on light backgrounds. I've gone through and started changing the rest of the token dashboard designs to make sure we're using the "-100" for text on light backgrounds since that darker color is compliant.

File: solidity/dashboard/src/components/Chip.jsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from "react"
+
+const Chip = ({
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply solidity/dashboard/src/components/Chip.jsx:3]: Its a component very similar to tags pls see https://www.figma.com/file/ARAMFLMOPal0JewjvlZGhX/Keep-Design-System?node-id=1952%3A7

Also, I wonder about a separate project e.g `keep-design` and registering it in npm. This package would contain all components that we use across web apps.

File: solidity/dashboard/src/components/CircularProgressBar.jsx [hunk 3]
```diff
@@ -19,37 +19,56 @@ const countProgressValue = (value, total, radius) => {
...
+  }, [value, total, normalizedRadius])
+
+  const center = parentSize ? parentSize / 2 : radius
+
   return (
-    <svg className="circular-progress-bar" width={120} height={120}>
+    <svg
+      className="circular-progress-bar"
+      width={parentSize || radius * 2}
+      height={parentSize || radius * 2}
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply solidity/dashboard/src/components/CircularProgressBar.jsx:51]: agree 👍 . I had no idea how to deal with it so I opted for the simplest and quickest solution that doesn't break other styles. I know is not too elegant. I'll take a look at it in spare cycles.

File: solidity/dashboard/src/components/TokenOverview.jsx [hunk 4]
```diff
@@ -4,90 +4,169 @@ import * as Icons from "./Icons"
...
+              : "No data to display"}
+          </h4>
+        )}
+      </div>
+    </section>
+  )
+}
+
+const TokenBalance = ({
+  tooltipProps,
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply solidity/dashboard/src/components/TokenOverview.jsx:106]: This will change when I replace obsolete tooltips with new ones.

**Comment by r-czajkowski** [comment]
> Reviewing here, @liz-shinn it looks like the balance _number_ should be darker (color-brand-mint-100) vs the keep logo (color-brand-mint-80), is that right?
> 
> Other things I noticed:
> 
>     * There's a hand cursor on the KEEP amount, and it's abbreviated (e.g. 562k), vs the designs which use the fully-expanded number (e.g. 562,000). There's a localization component here, which we can probably handle by calling `toLocaleString()` on the number to get the right separators for thousands and decimals.
> 
> 
> Going to have to accept the fact that I can't review 3k lines of changes haha. I left a few comments on a few things I noticed. Overall looks good, main flag is the Delegate buttons that are leading to a nonexistent URL (I believe it's delegate -> tokens/delegate).

Regarding token balance, we have an issue for that #2095. I going to fix all issues about `TokenAmount`  component in dedicated PR.

File: solidity/dashboard/src/components/TokenOverview.jsx (hunk 1 above)
Comment (r-czajkowski) [reply solidity/dashboard/src/components/TokenOverview.jsx:164]: fixed, pls see https://github.com/keep-network/keep-core/pull/2113/commits/0c060205b502b06f6c10551fcd6873cc55be082f

File: solidity/dashboard/src/components/Chip.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Shadowfiend) [reply solidity/dashboard/src/components/Chip.jsx:3]: I think we need things to settle a little more, but once that happens it seems like a strong idea.

**Pull request merged**